tests/integration/utils.py

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import hashlib
+import json
+import logging
+import os
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Dict, List, Tuple
+
+from git import Repo
+
+log = logging.getLogger(__name__)
+
+
+class TestParameters:
+    def __init__(
+        self,
+        *,
+        repo: str = "",
+        ref: str = "",
+        test_case: str = "",
+        packages: Tuple = (),
```

Comment:
Could we type-annotate the inner types as well? `tuple[<a>, ...]` for variable-length tuples, `tuple[<a>, <b>]` for fixed-length tuples, `list[<a>]`, ...